Makefile

```diff
@@ -26,13 +27,16 @@ update-readme:
 
 update: update-readme
 
-test-go:
+$(CODE_COVERAGE_REPORT_NAME_UNIT):
```

Comment:
This PR is about upgrading the mongo-driver as there is a moderate security alert, but I'm having hard times understanding all other changes in Makefile. Could you please explain why did you introduce this? Did you face any issue around these? The `CODE_COVERAGE_REPORT_NAME_UNIT` directory is created by the tool.

Reply:
I looked into this deeper and agree that there is an issue. Would you mind extracting it to a separate PR?

Reply:
Yeah, I encountered this while running the tests. I'll open a separate PR.